cpp/src/IceStorm/Grammar.y

```diff
@@ -1,37 +1,54 @@
-%{
+%code top{
 
 //
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
+}
+
+%code requires{
+
+// I must set the initial stack depth to the maximum stack depth to
+// disable bison stack resizing. The bison stack resizing routines use
+// simple malloc/alloc/memcpy calls, which do not work for the
+// YYSTYPE, since YYSTYPE is a C++ type, with constructor, destructor,
+// assignment operator, etc.
+#define YYMAXDEPTH  10000
+#define YYINITDEPTH YYMAXDEPTH
+
+// Newer bison versions allow to disable stack resizing by defining yyoverflow.
+#define yyoverflow(a, b, c, d, e, f) yyerror(a)
+
+}
+
+%code{
+
+// Forward declaration of the lexing function generated by flex, so bison knows about it.
+// This must match the definition of 'yylex' in the generated scanner.
+int yylex(YYSTYPE* yylvalp);
+
+}
+
+%{
```

Comment:
This code was moved from `Parser.h`. It's Bison specific code, so it had no business in `Parser.h`.

`requires` just means it's put in a specific spot in the header file, before Bison tries to define these things itself.